src/test/javascript/spec/component/orion/orion-exercise-details-student-actions.component.spec.ts

```diff
@@ -48,86 +40,88 @@ describe('OrionExerciseDetailsStudentActionsComponent', () => {
             providers: [
                 MockProvider(OrionBuildAndTestService),
                 MockProvider(OrionConnectorService),
-                MockProvider(FeatureToggleService),
                 { provide: ActivatedRoute, useValue: { queryParams: of({ withIdeSubmit: true }) } },
             ],
         })
-            .overrideModule(ArtemisTestModule, { set: { declarations: [], exports: [] } })
             .compileComponents()
             .then(() => {
                 fixture = TestBed.createComponent(OrionExerciseDetailsStudentActionsComponent);
                 comp = fixture.componentInstance;
                 orionConnector = TestBed.inject(OrionConnectorService);
-                forwardBuildSpy = spy(TestBed.inject(OrionBuildAndTestService), 'listenOnBuildOutputAndForwardChanges');
-                cloneSpy = spy(orionConnector, 'importParticipation');
-                submitSpy = spy(orionConnector, 'submit');
-                ideStateStub = stub(orionConnector, 'state');
+                forwardBuildSpy = jest.spyOn(TestBed.inject(OrionBuildAndTestService), 'listenOnBuildOutputAndForwardChanges');
+                cloneSpy = jest.spyOn(orionConnector, 'importParticipation');
+                submitSpy = jest.spyOn(orionConnector, 'submit');
+                ideStateStub = jest.spyOn(orionConnector, 'state');
             });
     });
+
     afterEach(() => {
-        sinon.restore();
+        jest.restoreAllMocks();
     });
 
     it('should not reflect that the represented exercise is opened if another exercise has been opened', fakeAsync(() => {
         const stateObservable = new BehaviorSubject(ideState);
         comp.exercise = exercise;
-        ideStateStub.returns(stateObservable);
+        ideStateStub.mockReturnValue(stateObservable);
 
         comp.ngOnInit();
         fixture.detectChanges();
         tick();
 
-        expect(comp.orionState.opened).to.not.equal(exercise.id);
+        expect(comp.orionState.opened).not.toBe(exercise.id);
```

Comment:
Should it be 40 instead (ideState.opened)? If so, please use that reference instead of the negation here.

Reply:
Yeah, but this test does not aim to check whether the id is stored correctly, but only checks whether it does not get overwritten by the other id. I agree this is not necessarily an edge case, but if I replace the comparison, it becomes the same test as the test below and can be removed. Would you prefer that?

Reply:
If I undestand it correctly with the second test you want to test that everything works with the same file and with the first test you want to test that if there is a mismatch, the mismatch gets overridden, correct?
I think in this case it makes sense to have two tests that test similar things. You could think about extracting them to a helper function to generify it a bit.
But neither would I keep it not would I remove them. Does this make sense to you?

Reply:
I did not write these tests initially, so my guess about the rationale is a good as yours.
Both tests check whether ngOnInit stores the orionState. Only in the one case the exercise id is the same as in the orionState and in the other it isn't. Looking at the code these are not different cases, though, since the values are stored in a totally different way anyways.

Reply:
Okay, well if it's really not different, then just remove one and check that the test description still fits.

Reply:
I now replaced it with the same test for the orion state thats present in all other components that subscribe to it